pkg/controller/proxyconfig/controller.go

```diff
@@ -0,0 +1,191 @@
+	}
+
+	// Fetch the proxy config
+	proxyConfig := &configv1.Proxy{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, proxyConfig)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			log.Println("proxy not found; reconciliation will be skipped", "request", request)
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, fmt.Errorf("failed to get proxy %q: %v", request, err)
+	}
+
+	// A nil proxy is generated by upgrades and installs not requiring a proxy.
+	if !isSpecHTTPProxySet(&proxyConfig.Spec) && !isSpecHTTPSProxySet(&proxyConfig.Spec) {
+		log.Println("httpProxy and httpsProxy not defined; reconciliation will be skipped", "request", request)
+		return reconcile.Result{}, nil
+	}
+
+	// Only proceed if the required config objects can be collected.
+	infraConfig := &configv1.Infrastructure{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get infrastructure config 'cluster': %v", err)
+	}
+	netConfig := &configv1.Network{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, netConfig); err != nil {
+		log.Printf("failed to get network config 'cluster': %v", err)
+		return reconcile.Result{}, err
+	}
+	clusterConfig := &corev1.ConfigMap{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: "cluster-config-v1", Namespace: "kube-system"}, clusterConfig); err != nil {
+		log.Printf("failed to get configmap 'cluster': %v", err)
+		return reconcile.Result{}, err
+	}
+
+	if err := r.ValidateProxyConfig(&proxyConfig.Spec); err != nil {
+		log.Printf("Failed to validate Proxy.Spec: %v", err)
+		r.status.SetDegraded(statusmanager.ProxyConfig, "InvalidProxyConfig",
+			fmt.Sprintf("The proxy configuration is invalid (%v). Use 'oc edit proxy.config.openshift.io cluster' to fix.", err))
+	}
+
+	// Update Proxy.config.openshift.io.Status
+	if err := r.syncProxyStatus(proxyConfig, infraConfig, netConfig, clusterConfig); err != nil {
+		log.Printf("Could not sync proxy status: %v", err)
+		r.status.SetDegraded(statusmanager.ProxyConfig, "StatusError",
+			fmt.Sprintf("Could not update proxy configuration status: %v", err))
+		return reconcile.Result{}, err
+	}
+	r.status.SetNotDegraded(statusmanager.ProxyConfig)
+	log.Printf("Reconciling Proxy.config.openshift.io %s complete\n", request.Name)
+
+	return reconcile.Result{}, nil
+}
+
+// isSpecHTTPProxySet checks whether spec.httpProxy of proxy is set.
+func isSpecHTTPProxySet(proxyConfig *configv1.ProxySpec) bool {
+	if len(proxyConfig.HTTPProxy) == 0 {
+		return false
+	}
+
+	return true
+}
+
+// isSpecHTTPSProxySet checks whether spec.httpsProxy of proxy is set.
+func isSpecHTTPSProxySet(proxyConfig *configv1.ProxySpec) bool {
```

Comment:
Maybe I should not even being looking at this but could we make it `return len(proxyConfig.HTTPSProxy) > 0`

Reply:
The same applies to other functions below. Feel free to just ignore my comments.

Reply:
`return len(proxyConfig.HTTPSProxy) > 0` did not provide the behavior I was looking for, so I reverted back in my latest commit.

Reply:
Weird. I have done this sample below and it seems like the behavior is the same. Could you point me in what cases it does not provide the behavior you expect? https://play.golang.org/p/opLt8dNrOKv

Reply:
I just came across the commit below, the patch is wrong as it uses `return len(proxyConfig.HTTPProxy) == 0` instead of `return len(proxyConfig.HTTPProxy) > 0`.



https://github.com/openshift/cluster-network-operator/commit/abf6019614e993a26ca20f56dec8b27d8e5157bb#diff-6cc3b6cf53aa0e5d6919e9943b4b174aR139

Reply:
Thanks for catching that. I've updated the PR.